core/src/main/java/org/testcontainers/containers/VncRecordingContainer.java

```diff
@@ -87,19 +88,27 @@ protected void configure() {
         );
     }
 
+    @SneakyThrows
+    public void saveRecordingToFile(File file) {
+        try(InputStream inputStream = streamRecording()) {
+            Files.copy(inputStream, file.toPath(), StandardCopyOption.REPLACE_EXISTING);
+        }
+    }
+
     @SneakyThrows
     public InputStream streamRecording() {
+        reencodeRecording();
+
         TarArchiveInputStream archiveInputStream = new TarArchiveInputStream(
                 dockerClient.copyArchiveFromContainerCmd(getContainerId(), RECORDING_FILE_NAME).exec()
         );
         archiveInputStream.getNextEntry();
         return archiveInputStream;
     }
 
-    @SneakyThrows
-    public void saveRecordingToFile(File file) {
-        try(InputStream inputStream = streamRecording()) {
-            Files.copy(inputStream, file.toPath(), StandardCopyOption.REPLACE_EXISTING);
-        }
+    private void reencodeRecording() throws IOException, InterruptedException {
+        String newFlvOutput = "/newScreen.flv";
```

Comment:
Why encode to `.flv` instead of `.mp4` container? Only to avoid the `-movflags faststart` parameter?

Reply:
I don't know if it's Ok to change the record extension in this ticket, maybe we can open a new ticket for this, and I add that we can implement a kind of configuration to let user decide which extension he wants for recording...

...but, I'm open if you want me to change my pull request to use `.mp4` instead of `.flv`  with `-movflags faststart`  :slightly_smiling_face:

Reply:
I personally have no opinion about this. 
@leonard84 could you elaborate why `.mp4` would be the better choice? Would be happy to endorse this instead and change it in this PR, if it is the better choice for whatever reason.

Reply:
> FLV is a proprietary format developed by Adobe. MP4 is a format standardized by the ISO in the MPEG-4 family. Your video is H.264 (MPEG-4 Part 10), so it's from the same MPEG-4 standard.2

>That being said, the support for MP4 should always be better than for FLV containers. While typical desktop video players support almost every container and codec, devices such as mobile phones or DLNA streaming servers might only handle MP4 files.

https://superuser.com/a/549647

Reply:
I can add this blog article of Adobe about `flv` being obsolete: 

> If you want to create a video to play in Flash Player, then you should generally be using H.264 (.mp4), not the obsolete FLV or F4V formats...


https://blogs.adobe.com/creativecloud/removal-flv-f4v-swf-export-adobe-media-encoder-after-effects-premiere-pro/

Reply:
IMHO we should offer `mp4` as default and could offer a legacy mode that produces `flv`. 
I don't think that there are any who specifically want `flv`, we currently transcode it in a post processing  step to `mp4`.
Sure this change breaks that, but that is the whole point, and we can just remove the step with this feature.

Reply:
Was this quietly resolved? I still think we should use `mp4`.
```suggestion
        String newFlvOutput = "/newScreen.mp4";
        execInContainer("ffmpeg" , "-i", RECORDING_FILE_NAME, "-vcodec", "libx264", "-movflags", "faststart", newFlvOutput);
```

Since the comment already has fixed line, I can't really let the suggestion replace the next line.